types/dialogflow/index.d.ts

```diff
@@ -17,6 +18,11 @@ export namespace v2 {
     getProjectId(): Promise<string>;
     getProjectId(callback?: (error: Error, id: string) => string): void;
 
+    setAgent(
```

Comment:
It is possible to pass the callback in place of options. See: https://github.com/googleapis/nodejs-dialogflow/blob/master/src/v2/agents_client.js#L358

Reply:
Would be easiest to simply provide some method overloading for `setAgent` rather than union types on the parameters.

Reply:
Updated

Reply:
I'll submit another PR to fix the rest of the methods that use this pattern as well. `getAgent`, `searchAgents`, `deleteAgent` should all use overloading.